src/lib/form-field/form-field.ts

```diff
@@ -86,6 +98,9 @@ let nextUniqueId = 0;
 export class MatFormField implements AfterViewInit, AfterContentInit, AfterContentChecked {
   private _labelOptions: LabelOptions;
 
+  /** The form-field style variant. */
+  @Input() variant: MatFormFieldVariant = 'legacy';
```

Comment:
Not necessarily in this PR, but we'll also need a provider to set the default appearance as well

Reply:
Yep, will add this later